[JIT] Fix returning instance of custom class from method

jamesr66a opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #32320 [JIT] Test passing custom class instance to bound method
* **#32312 [JIT] Fix returning instance of custom class from method**
* #32301 [JIT] Remove __torch__ from custom class qualname
* #32324 Temporary workaround for BC test due to schema parser changes

Differential Revision: [D19433511](https://our.internmc.facebook.com/intern/diff/D19433511)

Commit:
[JIT] Fix returning instance of custom class from method

[ghstack-poisoned]

kostmo commented:
## :pill: CircleCI build failures summary and remediations
As of commit 6658d8ed:


* **1/1** failures introduced in this PR


## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=6658d8ed740ecba88ff8e94e409b6192d8a8f564).

### :detective: 1 new failure recognized by patterns
The following build failures do not appear to be due to upstream breakage:
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/4278345) pytorch_linux_backward_compatibility_check_test (1/1)
**Step:** "Test" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=87295510) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=87295510))
<details>
<summary>
<code>Jan 16 21:01:48 RuntimeError: expected ident but found '.' here:</code>
</summary>

```
Jan 16 21:01:34 + pip install --progress-bar off --pre torch -f https://download.pytorch.org/whl/nightly/cpu/torch_nightly.html 
Jan 16 21:01:35 Looking in links: https://download.pytorch.org/whl/nightly/cpu/torch_nightly.html 
Jan 16 21:01:36 Collecting torch 
Jan 16 21:01:36   Downloading https://download.pytorch.org/whl/nightly/cpu/torch-1.5.0.dev20200116%2Bcpu-cp36-cp36m-linux_x86_64.whl (128.5MB) 
Jan 16 21:01:45 Installing collected packages: torch 
Jan 16 21:01:48 Successfully installed torch-1.5.0.dev20200116+cpu 
Jan 16 21:01:48 + python check_backward_compatibility.py --new-schemas new_schemas.txt 
Jan 16 21:01:48 Traceback (most recent call last): 
Jan 16 21:01:48   File "check_backward_compatibility.py", line 132, in <module> 
Jan 16 21:01:48     s = parse_schema(line.strip()) 
Jan 16 21:01:48 RuntimeError: expected ident but found '.' here: 
Jan 16 21:01:48 _TorchScriptTesting_StackString::clone(torch.classes._TorchScriptTesting_StackString _0) -> (torch.classes._TorchScriptTesting_StackString _0) 
Jan 16 21:01:48                                             ~ <--- HERE 
Jan 16 21:01:48 =================== sccache compilation log =================== 
Jan 16 21:01:48 + cleanup 
Jan 16 21:01:48 + retcode=1 
Jan 16 21:01:48 + set +x 
Jan 16 21:01:48 =========== If your build fails, please take a look at the log above for possible reasons =========== 
Jan 16 21:01:48 Compile requests                 0 
Jan 16 21:01:48 Compile requests executed        0 
Jan 16 21:01:48 Cache hits                       0 
```

</details>



---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

Commit:
Update on "[JIT] Fix returning instance of custom class from method"


Differential Revision: [D19433511](https://our.internmc.facebook.com/intern/diff/D19433511)

[ghstack-poisoned]

Commit:
Update on "[JIT] Fix returning instance of custom class from method"


Differential Revision: [D19433511](https://our.internmc.facebook.com/intern/diff/D19433511)

[ghstack-poisoned]

Commit:
Update on "[JIT] Fix returning instance of custom class from method"


Differential Revision: [D19433511](https://our.internmc.facebook.com/intern/diff/D19433511)

[ghstack-poisoned]

Commit:
Update on "[JIT] Fix returning instance of custom class from method"


Differential Revision: [D19433511](https://our.internmc.facebook.com/intern/diff/D19433511)

[ghstack-poisoned]

Commit:
Update on "[JIT] Fix returning instance of custom class from method"


Differential Revision: [D19433511](https://our.internmc.facebook.com/intern/diff/D19433511)

[ghstack-poisoned]

Commit:
Update on "[JIT] Fix returning instance of custom class from method"


Differential Revision: [D19433511](https://our.internmc.facebook.com/intern/diff/D19433511)

[ghstack-poisoned]
